### PR TITLE
Enable dry-run e2e tests for historical versions

### PR DESCRIPTION
**Pull request by SataQiu**
`kubeadm` {init | join | upgrade | reset}  have a `--dry-run` flag since v1.24 ([#107512](https://github.com/kubernetes/kubernetes/pull/107512)).
So, it is safe to turn on dryrun e2e tests for more historical versions now.

test-infra PR: https://github.com/kubernetes/test-infra/pull/28581

### CODE REVIEW

File: kinder/ci/tools/update-workflows/config.yaml
Comment (pacoxu) [inline kinder/ci/tools/update-workflows/config.yaml:191]: https://github.com/kubernetes/kubeadm/blob/2c55ced7317ae3def1167775d2dfb0253a403e67/kinder/ci/workflows/dryrun-tasks.yaml#L11

There is a var `upgradeVersion` in dryrun-tasks.yaml.
Should we set it as well?

Comment (SataQiu) [reply kinder/ci/tools/update-workflows/config.yaml:193]: Yeah, we should set it. Only the `latest` Kubernetes version does not need to be configured.

**Review by pacoxu — APPROVED** [review]
/lgtm

**Comment by SataQiu** [comment]
waiting for more review :)

**Comment by SataQiu** [comment]
cc @neolit123 @pacoxu 
Need lgtm label

File: kinder/ci/tools/update-workflows/templates/workflows/dryrun-tasks.yaml
Comment (pacoxu) [inline kinder/ci/tools/update-workflows/templates/workflows/dryrun-tasks.yaml:105]: Should we add `/kinder/upgrade/{{ .vars.upgradeVersion }}/` in below reset task as well?

Line 112-113.

Comment (SataQiu) [reply kinder/ci/tools/update-workflows/templates/workflows/dryrun-tasks.yaml:105]: No, we not really do upgrade, just dry-run here.

Comment (pacoxu) [reply kinder/ci/tools/update-workflows/templates/workflows/dryrun-tasks.yaml:105]: Got it. (forget that this is for dry-run.)

**Pull request merged**
